mne/io/boxy/boxy.py

```diff
@@ -234,27 +295,32 @@ def __init__(self, fname, preload=False, verbose=None):
                                     fiducial_coords[1], 
                                     fiducial_coords[2])
         
-        for i_chan in range(len(boxy_labels)-1):
+        for i_chan in range(len(boxy_labels)):
             temp_ch_src_det = apply_trans(native_head_t, boxy_coords[i_chan][:9].reshape(3, 3)).ravel()
             temp_other = np.asarray(boxy_coords[i_chan][9:], dtype=np.float64) # add wavelength and placeholders
-            info['chs'][i_chan]['loc'] = np.concatenate((temp_ch_src_det, temp_other), axis=0)
-        info['chs'][-1]['loc'] = np.zeros((12,))   #remove last line?     
-        
+            info['chs'][i_chan]['loc'] = np.concatenate((temp_ch_src_det, temp_other), axis=0) 
+                
         raw_extras = {'source_num': source_num,
                      'detect_num': detect_num, 
                      'start_line': start_line,
                      'end_line': end_line,
                      'filetype': filetype,
-                     'files': files,}
-
-        print('Start Line: ', start_line)
-        print('End Line: ', end_line)
-        print('Original Difference: ', end_line-start_line)
-        first_samps = start_line
+                     'files': files,
+                     'data_types': data_types,}
+        
+        print('Start Line: ', start_line[0])
+        print('End Line: ', end_line[0])
+        print('Original Difference: ', end_line[0]-start_line[0])
+        first_samps = start_line[0]
         print('New first_samps: ', first_samps)
-        diff = end_line-start_line
+        diff = end_line[0] - start_line[0]
+
         #input file has rows for each source, output variable rearranges as columns and does not
-        last_samps = start_line + diff // source_num -1 
+        if filetype[0] == 'non-parsed':
```

Comment:
are we still supporting both of these, if not, lets simplify by removing unneeded

Reply:
yeah the code should be able to handle both. Figure it keeps the code flexible and then people don't have to worry about saving the boxy data in a specific way.

Reply:
ok sounds good, if the code is ment to handle both we want another dataset in another folder and want to be running tests on it as well

Reply:
okay, I'll add some extra tests. I think all the old datasets are non-parsed and the only parsed file I have is from running trigger tests.

I was using .mtg and .elp files from another dataset to test the parsed file, but I'll make sure it stills works after all the recent changes.